iroha/src/tx.rs

```diff
@@ -71,33 +73,43 @@ impl AcceptedTransaction {
         let mut world_state_view_temp = world_state_view.clone();
         let account_id = self.payload.account_id.clone();
         if !is_genesis && account_id == <Account as Identifiable>::Id::genesis_account() {
-            return Err(
-                self.reject("Genesis account can sign only transactions in the genesis block.")
-            );
+            return Err(self.reject(PipelineRejectionReason::GenesisAccountSign));
         }
         for signature in &self.signatures {
-            if let Err(e) = signature.verify(self.hash().as_ref()) {
+            if let Err(reason) = signature.verify(self.hash().as_ref()) {
+                let signature = signature.clone();
                 return Err(self
                     .clone()
-                    .reject(&format!("Failed to verify signatures: {}", e)));
+                    .reject(PipelineRejectionReason::VerifySignatures { signature, reason }));
             }
         }
         if !self
             .check_signature_condition(world_state_view)
-            .map_err(|error| self.clone().reject(&error))?
+            .map_err(|reason| PipelineRejectionReason::SatisfySignatureCondition { reason })
+            .map_err(|reason| self.clone().reject(reason))?
         {
-            return Err(self.reject("Signature condition not satisfied."));
+            let reason = "Signature condition not satisfied.".to_owned();
+            return Err(self.reject(PipelineRejectionReason::SatisfySignatureCondition { reason }));
```

Comment:
Duplicated information here. It can be just an empty enum variant for now without `reason`.